frame/identity/src/lib.rs

```diff
@@ -474,18 +474,28 @@ decl_module! {
 		/// - One storage mutation (codec `O(R)`).
 		/// - One event.
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(MINIMUM_WEIGHT)]
-		fn add_registrar(origin, account: T::AccountId) {
+		#[weight = FunctionOf(
+			|(_, &old_count): (&T::AccountId, &u32)| {
+				T::DbWeight::get().reads_writes(1, 1)
+				+ 500_000 * (old_count as u64)
+			},
+			DispatchClass::Normal,
+			true
+		)]
+		fn add_registrar(origin, account: T::AccountId, old_registrar_count: u32) -> DispatchResult {
 			T::RegistrarOrigin::try_origin(origin)
 				.map(|_| ())
 				.or_else(ensure_root)?;
+			let mut registrars = Self::registrars();
+			ensure!(registrars.len() as u32 <= old_registrar_count, "invalid count");
 
-			let i = <Registrars<T>>::mutate(|r| {
-				r.push(Some(RegistrarInfo { account, fee: Zero::zero(), fields: Default::default() }));
-				(r.len() - 1) as RegistrarIndex
-			});
+			registrars.push(Some(RegistrarInfo { account, fee: Zero::zero(), fields: Default::default() }));
+			let i = (registrars.len() - 1) as RegistrarIndex;
+			<Registrars<T>>::put(registrars);
```

Comment:
Gav had commented on one of my previous PRs that we should prefer `try_mutate`

See here: https://github.com/paritytech/substrate/pull/5314#discussion_r396030969

Reply:
There is no `try_mutate` on regular storage items AFAIK

Reply:
Good catch, made a PR: https://github.com/paritytech/substrate/pull/5699

Reply:
it has been merged